### PR TITLE
Refactor - Better Error Handling / Surfacing.

### PR DESCRIPTION
**Pull request by pkluz**
@allewun Requesting expedited review here. It's a blocking issue :)

### CODE REVIEW

File: Pod/Classes/OriginateHTTPClient.m [hunk 1]
```diff
@@ -440,51 +306,73 @@ + (void)DELETEResource:(NSString *)URI
...
 
-
-#pragma mark - Empty HTTP body
++ (void)handleResponse:(NSURLResponse *)response
+                  data:(NSData *)data
+                 error:(NSError *)responseError
+evaluateLocationHeader:(BOOL)evaluateLocationHeader
+            completion:(OriginateHTTPClientResponse)responseBlock
+{
+    NSHTTPURLResponse *HTTPResponse = (NSHTTPURLResponse *)response;
```
Comment (pkluz) [inline Pod/Classes/OriginateHTTPClient.m:325]: @allewun Take a look at this section specifically to make sure I covered all the bases

File: Pod/Classes/OriginateHTTPClient.m [hunk 2]
```diff
@@ -569,41 +432,38 @@ + (NSError *)errorTimeout
...
-
-+ (NSError *)HTTPError500InternalServer
-{
-    return [NSError errorWithDomain:[[self class] errorDomain]
-                               code:500
-                           userInfo:@{ NSLocalizedDescriptionKey : @"The server encountered an unexpected condition which prevented it from fulfilling the request." }];
+    switch (code) {
+        case 400:
+            return @"The request could not be understood by the server due to malformed syntax.";
+            break;
```
Comment (allewun) [inline Pod/Classes/OriginateHTTPClient.m]: remove `break` (or add them to all, I'd prefer removal but up to you)

**Comment by allewun** [comment]
`errorTimeout` doesn't seem to be used anymore. Is this intentional?

**Comment by pkluz** [comment]
Good find on the timeout

**Pull request closed without merging**